plugins/obs-nvenc/nvenc.c

```diff
@@ -380,34 +382,47 @@ static bool init_encoder_base(struct nvenc_data *enc, obs_data_t *settings)
 	config->rcParams.qpMapMode = NV_ENC_QP_MAP_DELTA;
 
 	/* -------------------------- */
-	/* initialize                 */
-
-	info("settings:\n"
-	     "\tcodec:        %s\n"
-	     "\trate_control: %s\n"
-	     "\tbitrate:      %d\n"
-	     "\tmax_bitrate:  %d\n"
-	     "\tcq/cqp:       %ld\n"
-	     "\tkeyint:       %d\n"
-	     "\tpreset:       %s\n"
-	     "\ttuning:       %s\n"
-	     "\tmultipass:    %s\n"
-	     "\tprofile:      %s\n"
-	     "\twidth:        %d\n"
-	     "\theight:       %d\n"
-	     "\tb-frames:     %ld\n"
-	     "\tb-ref-mode:   %ld\n"
-	     "\tlookahead:    %s (%d)\n"
-	     "\taq:           %s\n"
-	     "\tsplit encode: %ld\n"
-	     "\tuser opts:    %s\n",
-	     get_codec_name(enc->codec), enc->props.rate_control, bitrate,
-	     max_bitrate, vbr ? enc->props.target_quality : enc->props.cqp,
-	     gop_size, enc->props.preset, enc->props.tune, enc->props.multipass,
-	     enc->props.profile, enc->cx, enc->cy, enc->props.bf,
-	     enc->props.bframe_ref_mode, lookahead ? "true" : "false",
-	     rc_lookahead, enc->props.adaptive_quantization ? "true" : "false",
-	     enc->props.split_encode, enc->props.opts_str);
+	/* log settings	              */
+
+	struct dstr log = {0};
+
+	dstr_catf(&log, "\tcodec:        %s\n", get_codec_name(enc->codec));
+	dstr_catf(&log, "\trate_control: %s\n", enc->props.rate_control);
+
+	if (bitrate && !cqvbr)
+		dstr_catf(&log, "\tbitrate:      %d\n", bitrate);
+	if (vbr)
+		dstr_catf(&log, "\tmax_bitrate:  %d\n", max_bitrate);
+	if (cqp)
+		dstr_catf(&log, "\tcqp:          %ld\n", enc->props.cqp);
+	if (cqvbr) {
+		dstr_catf(&log, "\tcq:           %ld\n",
+			  enc->props.target_quality);
+	}
```

Comment:
The only minor annoyance here is that, due to the indentation, it is difficult to see visually that all the spacing lines up correctly. Maybe some day, we'll just make a convenience function that does this, or we'll replace the logging system entirely with something that does this. This is not a blocking concern though, just wanted to mark it.